### PR TITLE
LibJS: Small fix for spreading, optimisation for indexed properties

### PR DESCRIPTION
**Pull request by davidot**
And also fix an include style

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp [hunk 1]
```diff
@@ -35,8 +35,13 @@ void SimpleIndexedPropertyStorage::grow_storage_if_needed()
 {
     if (m_array_size <= m_packed_elements.size())
         return;
-    // Grow storage by 25% at a time.
-    m_packed_elements.resize(m_array_size + (m_array_size / 4));
+
+    if (m_array_size <= m_packed_elements.capacity()) {
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp:39]: Usually these allow up to ~75-80% capacity utilisation before growing again, maybe experiment with that too?
(not a requirement or anything, just curious)

File: Userland/Libraries/LibJS/AST.cpp [hunk 2]
```diff
@@ -2987,28 +2987,10 @@ Completion ObjectExpression::execute(Interpreter& interpreter, GlobalObject& glo
...
-                    }
-                }
-            } else if (key.is_string()) {
-                auto& str_to_spread = key.as_string().string();
-
-                for (size_t i = 0; i < str_to_spread.length(); i++)
-                    object->define_direct_property(i, js_string(interpreter.heap(), str_to_spread.substring(i, 1)), JS::default_attributes);
-            }
+            // 4. Return ? CopyDataProperties(object, fromValue, excludedNames).
+            TRY(object->copy_data_properties(key, {}, global_object));
```
Comment (alimpfard) [inline Userland/Libraries/LibJS/AST.cpp:2993]: nice!

**Review by alimpfard — APPROVED** [review]
just some thoughts, nothing blocking (changes LGTM)

**Review by linusg — APPROVED** [review]
LGTM, very cool!

File: Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp (hunk 1 above)
Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp:39]: From my testing, this doesn't seem to improve performace. Although I might not quite understand what you mean. Why exactly would we want to always keep ~20% free?

Comment (alimpfard) [reply Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp:39]: I don't really have a concrete idea why, I've just seen it done a lot 😅

Comment (davidot) [reply Userland/Libraries/LibJS/Runtime/IndexedProperties.cpp:39]: We are still a lot slower than v8 and sm but I think we can leave this until we have more suitable benchmarks (I bet they detect this kind of push in a loop anyway)

**Pull request merged**